tests/hardhat/lens/RewardsSummary.ts

```diff
@@ -0,0 +1,184 @@
+  rewardDistributor3.rewardTokenSupplyState.returns({
+    index: convertToUnit(1, 18),
+    block: 1,
+  });
+
+  vBUSD.borrowIndex.returns(convertToUnit(1, 18));
+  vBUSD.totalBorrows.returns(convertToUnit(10000, 8));
+  vBUSD.totalSupply.returns(convertToUnit(10000, 8));
+  vBUSD.balanceOf.returns(convertToUnit(100, 8));
+  vBUSD.borrowBalanceStored.returns(convertToUnit(100, 8));
+
+  vWBTC.borrowIndex.returns(convertToUnit(1, 18));
+  vWBTC.totalBorrows.returns(convertToUnit(100, 18));
+  vWBTC.totalSupply.returns(convertToUnit(100, 18));
+  vWBTC.balanceOf.returns(convertToUnit(100, 8));
+  vWBTC.borrowBalanceStored.returns(convertToUnit(100, 8));
+
+  return {
+    comptroller,
+    vBUSD,
+    vWBTC,
+    rewardDistributor1,
+    rewardToken1,
+    rewardDistributor2,
+    rewardToken2,
+    rewardDistributor3,
+    rewardToken3,
+    poolLens,
+  };
+};
+
+describe("PoolLens: Rewards Summary", () => {
+  beforeEach(async () => {
+    [account] = await ethers.getSigners();
+    ({
+      comptroller,
+      vBUSD,
+      vWBTC,
+      rewardDistributor1,
+      rewardToken1,
+      rewardDistributor2,
+      rewardToken2,
+      rewardDistributor3,
+      rewardToken3,
+      poolLens,
+    } = await loadFixture(rewardsFixture));
+  });
+  it("Should get summary for all markets", async () => {
+    // Mine some blocks so deltaBlocks != 0
+    await mine(11);
+
+    const accountAddress = await account.getAddress();
+
+    await poolLens.getPendingRewards(accountAddress, comptroller.address);
+    expect(comptroller.getAllMarkets).to.have.been.calledOnce;
+    expect(comptroller.getRewardDistributors).to.have.been.calledOnce;
+
+    expect(rewardDistributor1.rewardToken).to.have.been.calledOnce;
+    expect(rewardDistributor2.rewardToken).to.have.been.calledOnce;
+    expect(rewardDistributor3.rewardToken).to.have.been.calledOnce;
+
+    expect(rewardDistributor1.rewardTokenAccrued).to.have.been.calledOnce;
+    expect(rewardDistributor2.rewardTokenAccrued).to.have.been.calledOnce;
+    expect(rewardDistributor3.rewardTokenAccrued).to.have.been.calledOnce;
+
+    // Should be called once per market
+    expect(rewardDistributor1.rewardTokenBorrowState).to.have.been.callCount(2);
+    expect(rewardDistributor2.rewardTokenBorrowState).to.have.been.callCount(2);
+    expect(rewardDistributor3.rewardTokenBorrowState).to.have.been.callCount(2);
+
+    expect(rewardDistributor1.rewardTokenSupplyState).to.have.been.callCount(2);
+    expect(rewardDistributor2.rewardTokenSupplyState).to.have.been.callCount(2);
+    expect(rewardDistributor3.rewardTokenSupplyState).to.have.been.callCount(2);
+
+    // Should be called once per reward token configured
+    expect(vBUSD.borrowIndex).to.have.been.callCount(3);
+    expect(vWBTC.borrowIndex).to.have.been.callCount(3);
+
+    // Should be called once per reward token configured
+    expect(vBUSD.totalBorrows).to.have.been.callCount(3);
+    expect(vWBTC.totalSupply).to.have.been.callCount(3);
+  });
```

Comment:
Shouldn't we check the values in the response? Given the prepared set-up and the number of blocks I would say we should be able to know the expected values.

Reply:
Yeah I included a check for the expected return value